### PR TITLE
CLI: SharePoint --data flag and wildcards

### LINKED ISSUE #2579: CLI: SharePoint: Libraries Select `*` with data constraint INOP
`./corso backup create sharepoint --web-url tenant.sharepoint.com --data libraries` CORRECT
`./corso backup create sharepoint --web-url ‘*’ --data libraries` **INCORRECT**
    - Incorrect version: Attempts to back up all categories of SharePoint`pages, lists, and libraries`.

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Enables feature SharePoint to use "wildcard" to select all available sites with the ability to only backup selected categories.

Example 
```bash
./corso backup create sharepoint --web-url "*" --data libraries
```

## Does this PR need a docs update or release note?
- [x] :no_entry: No 

## Type of change

- [x] :bug: Bugfix

## Issue(s)

* closes #2579<issue>

## Test Plan

- [x] :muscle: Manual

### CODE REVIEW

File: src/cli/backup/sharepoint.go [hunk 1]
```diff
@@ -334,6 +328,21 @@ func sharePointBackupCreateSelectors(
 		return sel, nil
 	}
 
+	return addCategories(sel, cats), nil
+}
+
+func handleWildCard(categories []string) *selectors.SharePointBackup {
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:334]:
```suggestion
func includeAllSitesWithCategories(categories []string) *selectors.SharePointBackup {
```

File: src/cli/backup/sharepoint.go [hunk 2]
```diff
@@ -334,6 +328,21 @@ func sharePointBackupCreateSelectors(
...
+	sel := selectors.NewSharePointBackup(selectors.Any())
+	if len(categories) == 0 {
+		sel.Include(sel.AllData())
+	} else {
+		sel = addCategories(sel, categories)
+	}
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:340]: The len check should either get moved into `addCategories`, or the addCategories behavior should fold into `handleWildCard`.

**Review by ryanfkeepers — APPROVED** [review]

File: src/cli/backup/sharepoint.go (hunk 2 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved within [3b162ac](https://github.com/alcionai/corso/pull/2580/commits/3b162ac9588ee1f1ece006323b5490b5f29bf284)

File: src/cli/backup/sharepoint.go (hunk 1 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved within [3b162ac](https://github.com/alcionai/corso/pull/2580/commits/3b162ac9588ee1f1ece006323b5490b5f29bf284)

**Pull request merged**